src/TorchSharp/NN/Module.cs

```diff
@@ -512,14 +515,40 @@ public virtual bool has_parameter(string target)
                     if (_internal_params.TryGetValue(target, out var parameter)) {
                         return true;
                     }
-                    foreach (var child in named_children().Where(nc => target.StartsWith(nc.name))) {
-                        var prefix = child.name + ".";
-                        if (child.module.has_parameter(target.Remove(0, prefix.Length)))
-                            return true;
+
+                    var splits = target.Split('.');
+                    if (splits.Length > 1) {
+                        foreach (var child in named_children().Where(nc => nc.name == splits[0])) {
+                            if (child.module.has_parameter(target.Remove(0, splits[0].Length + 1)))
+                                return true;
+                        }
                     }
                     return false;
                 }
 
+                /// <summary>
+                /// Returns the buffer given by target if it exists, otherwise throws an error.
+                /// </summary>
+                /// <param name="target">The fully-qualified string name of the buffer to look for.</param>
+                /// <returns>The tensor referenced by target</returns>
+                public virtual Tensor get_buffer(string target)
+                {
+                    if (target is null) throw new ArgumentNullException("target");
+                    if (_internal_buffers.TryGetValue(target, out var buffer)) {
+                        return buffer;
+                    }
+
+                    var splits = target.Split('.');
+                    if (splits.Length > 1) {
+                        foreach (var child in named_children().Where(nc => nc.name == splits[0])) {
+                            var p = child.module.get_buffer(target.Remove(0, splits[0].Length + 1));
+                            if (p is not null)
+                                return p;
+                        }
+                    }
+                    return null;
+                }
```

Comment:
NIT: this repeats a bit. Perhaps make a `T get_named<T>(name, internal_collection, getter)`?